libraries/AP_HAL_Linux/RCOutput_Sysfs.cpp

```diff
@@ -47,11 +47,6 @@ void RCOutput_Sysfs::init()
             AP_HAL::panic("RCOutput_Sysfs_PWM: Unable to setup PWM pin.");
         }
         _pwm_channels[i]->enable(false);
-
-        /* Set the initial frequency */
-        _pwm_channels[i]->set_freq(50);
-        _pwm_channels[i]->set_duty_cycle(0);
```

Comment:
I think the right fix would be to reorder set_freq and set_duty_cycle. According to what you are saying it seems that it would set the frequency and then whatever the duty cycle was will remain.

In my case I do a couple of things: 1) I have a OE which is set to 1 before init and 2) have a ExecStop in the systemd unit file that sets the OE again to 1 whenever the process exit (either clean or with an error).